arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -0,0 +1,199 @@
+        # section, we want to return key:value pairs for each of the IRQ, Thr, and Usr
+        # total stats.
+
+        # $ sudo rtla timerlat hist -u -d 1 -c 1,2 -E 10 -b 10
+        # # RTLA timerlat histogram
+        # # Time unit is microseconds (us)
+        # # Duration:   0 00:00:02
+        # Index   IRQ-001   Thr-001   Usr-001   IRQ-002   Thr-002   Usr-002
+        # 0          1000       944       134       999       998       986
+        # 10            0        55       863         0         1        13
+        # 20            0         1         3         0         0         0
+        # over:         0         0         0         0         0         0
+        # count:     1000      1000      1000       999       999       999
+        # min:          0         3         4         0         2         3
+        # avg:          1         7        10         0         2         3
+        # max:          7        21        24         3        11        15
+        # ALL:        IRQ       Thr       Usr
+        # count:     1999      1999      1999
+        # min:          0         2         3
+        # avg:          0         4         6
+        # max:          7        21        24
+
+        time_unit = ""
+        col_headers = []
+        latency_hist = []
+        total_irq_latency = {}
+        total_thr_latency = {}
+        total_usr_latency = {}
+        stats_names = ["over:", "count:", "min:", "avg:", "max:"]
+        stats_per_col = []
+        found_all = False
+
+        re_isunit = re.compile(r"^#\ Time\ unit")
+        re_isindex = re.compile(r"^Index")
+        re_isdigit = re.compile(r"^\d")
+        re_isall = re.compile(r"^ALL")
+
+        for line in output.splitlines():
+            if re_isunit.match(line):
+                time_unit = line.split()[4]
+            # Capture the column headers
+            elif re_isindex.match(line):
+                col_headers = line.lower().split()
+            # Stats names repeat, so flag when have passed ^ALL
+            elif re_isall.match(line) and not found_all:
+                found_all = True
+            # Either this is a histogram bucket row, or the first time we have seen
+            # a row beginning with a stat name
+            elif (re_isdigit.match(line)) or (
+                line.split()[0] in stats_names and not found_all
+            ):
+                # Capture the columnar data
+                row_obj = {}
+                for i, col in enumerate(col_headers):
+                    row_obj[col] = line.split()[i]
+                if re_isdigit.match(line):
+                    latency_hist.append(row_obj)
+                else:
+                    stats_per_col.append(row_obj)
+            # This is the second time we have seen a row beginning with a stat name, so
+            # we want to generate key:value pairs instead of columnar data
+            elif found_all and line.split()[0] in stats_names:
```

Comment:
The phrase "This is the second time we have seen a row beginning with a stat name" might be accurate (depending on the actually output), but it doesn't actually reflect _the code_, so I recommend changing this text to something like:
> Since we've encountered the summary statistics (marked by the line starting with "ALL"), generate key:value pairs instead of columnar data.